### DISCUSSION TITLE
Webradios

### DISCUSSION THREAD

**Discussion by opengk** [jcorporation/myMPD#769] [Q&A] [answered]
Hi,

After updating to 9.3.2 under Archlinux, and rebooting I lost the webradios listing.

Did I miss something ? Is there anythong to do to get webradios or does it come out of the box when mympd is installed ?

Kind regards.

mympd version 9.3.3
mpd v0.23.7
Linux version 5.17.1-1-aarch64-ARCH
On raspberry pi 4

**Comment by jcorporation** [comment]
I did not understand exactly what your problem is.

- are the webradio favorites lost after upgrade?
- Upgrade from which version?

There is nothing to configure for webradios. Webradio favorites are saved in the folder /var/lib/mympd/webradios.

**Comment by opengk** [comment]
The upgrade was from 9.2 to 9.3 versions.

Indeed, the webradios were present with 9.2 and do not appear anymore in 9.3

And the webradio folder is empty, that's strange.

Radio-browser.info works but not webradiodb. It's empty.

**Comment by jcorporation** [comment]
Webradio folder is only for favorites, not for webradiodb. Any error in the logs? The webradiodb is fetched on demand from GitHub. Sometimes the fetch works but nothing is displayed. Go back to favorites and open webradiodb fixes it.

**Comment by opengk** [comment, marked as answer]
Okay I understand better.

1. I look into radio-browser.info
2. When radio of choice is found, I have to add it to webrdios favorites
3. This is indeed creating an entry into /var/lib/mympd/webradios/

Is it how it's supposed to work ?

However I still don't find webradios when looking into webradiodb instead of radio-browser.info

**Comment by opengk** [comment]
> Webradio folder is only for favorites, not for webradiodb. Any error in the logs? The webradiodb is fetched on demand from GitHub. Sometimes the fetch works but nothing is displayed. Go back to favorites and open webradiodb fixes it.

We just replied at the same time... Let me try this and look at the logs.

**Comment by opengk** [comment]
So, I stopped the service, set logging to level 7, started mympd from console, and magic made webradiodb radios appear. No error into logs...

After stopping by ctrl C, and starting the service back with systemctl, it still works fine.

Issue solved. Thanks for the assistance.

Unfortunately I have no idea why this issue happened.

By the way, I dont't know if I missed something in the documentation, but in case it's missing, adding some dedicated doc about webradio favourites that are saved in /var/lib/mympd/webradios as single file would be great. Also when using favourites, the presentation with images is really cool. If it can help somebody else...

Thanks again, I like your app !

**Comment by jcorporation** [comment]
`/var/lib/mympd/webradios` is a directory. Webradio favorites are saved as individual m3u files there. 

The webradiodb is a single json file, fetched from the webradiodb repository and cached in the `/var/cache/mympd/webradio` folder. Lifetime is one day.

I do not think that this is documented. More documentation will be always better. Everbody could help too improve it ;)
